tests/lib/agent/Agent.test.ts

```diff
@@ -55,14 +57,13 @@ describe('Agent class', () => {
 
     test('can register node', async () => {
       const nonAgentNodePath = path.join(tempDir, `SomePolykey-${randomString()}`)
-      const pk = new Polykey(nonAgentNodePath, fs)
-      await pk.keyManager.generateKeyPair('John Smith', 'john@email.com', 'very password', 1024, true)
+      const km = new KeyManager(nonAgentNodePath, fs)
```

Comment:
Unit testing polykey should be structured like this. Once you have broken up the interfaces and developed the core models.

Create unit tests for just the core models and few compositions.

However for the rest of the system, we need to use `xstate-test` and `fast-check`.

Something like must be done with model-based testing and property-based testing. So you should start on this immediately afterwards as we will need that for Introcept too.

Reply:
yeah I've been trying to really break them up so they can be mocked. Need to get it right in the cleanup.